### PR TITLE
Fixes #44 and #17; Add eslint.

### LINKED ISSUE #44: ECMA6 Features Break jslint Task
Since we only run the linter through the react compiler (and not the full babel compiler), ECMA6 features break the task.

### PR DESCRIPTION
**Pull request by codeviking**
@jkinkead Please review.

This replaces the hack where JSX was being compiled prior to being linted (making the linter effectively useless).

FYI @sbhaktha Once this merges we should bump the demo to use this.  A jslinter will be nice to have (and I'd love it if you'd try it out and help me iron out the configuration, as it might be too noisy / not noisy enough right now)

### CODE REVIEW

File: tests/gulp-test.js [hunk 1]
```diff
@@ -204,22 +204,6 @@ describe('syrup.gulp.init()', function() {
...
-      captured.push(Array.prototype.slice.apply(arguments));
-    };
-    gulp.start('jslint', function() {
-      // Revert the capture
-      console.log = orig;
-      assert(captured.join('').indexOf('2 warnings') !== -1);
-      done();
-    })
-  });
-
```
Comment (codeviking) [inline tests/gulp-test.js]: `gulp-eslint` (kindly) logs it's output to `gutil.log` which isn't as easy to capture and accordingly test -- so I removed the unit test.

File: dev/app/main.jsx [hunk 2]
```diff
@@ -1,6 +1,4 @@
-'use strict';
```
Comment (jkinkead) [inline dev/app/main.jsx]: Does this not do anything?

File: gulp/index.js [hunk 3]
```diff
@@ -332,12 +331,12 @@ module.exports = {
      */
     gulp.task('jslint', function() {
       if (!options.disableJsHint) {
-        gutil.log(util.format('Linting javascript: %s', gutil.colors.magenta(paths.jshint)));
-        return gulp.src(paths.jshint)
+        gutil.log(util.format('Linting javascript: %s', gutil.colors.magenta(paths.jsLint)));
+        return gulp.src(paths.jsLint)
```
Comment (jkinkead) [inline gulp/index.js]: I assume the paths include `.jsx` files?

**Comment by jkinkead** [comment]
Just a couple of questions - LGTM! Merge when you're ready.

File: gulp/index.js (hunk 3 above)
Comment (codeviking) [inline gulp/index.js]: Yup!

File: dev/app/main.jsx (hunk 2 above)
Comment (codeviking) [inline dev/app/main.jsx]: Turns out `strict` mode is default for modules -- so we don't need to do this anymore.

**Pull request merged**